lib/Parse/Lexer.cpp

```diff
@@ -1884,23 +1884,12 @@ void Lexer::lexStringLiteral(unsigned CustomDelimiterLen) {
         // Successfully scanned the body of the expression literal.
         ++CurPtr;
         continue;
+      } else if (!isMultilineString) {
+        diagnose(CurPtr, diag::lex_unterminated_string);
+        diagnose(CurPtr, diag::string_interpolation_unclosed);
```

Comment:
This needs to return `formToken(tok::unknown, TokStart)` like we did for regular string literals under the `*CurPtr == '\r' || *CurPtr == '\n'` condition.